### DISCUSSION TITLE
Need for a tree data structure

### DISCUSSION THREAD

**Discussion by sandwoodK** [nushell/nushell#6680] [Ideas]
Some data out there are fondamentaly organized as tree. 

Nushell visualisation of such data is not good. 

Open an xml file for example.   Obviously the proposed data organization in nested records and tables is not usefull for an end-user. 
filter ,sorting and other functions of nushell are barely usable because of the nature of the proposed data organisation.

We would need at least a tree view (if we can't have a real tree data) and some tree filter facility ( i.e "give me al the node at depth 3".

**Comment by fdncred** [comment]
Nushell used to have a tree command but I think it was for the filesystem, iirc. It would be nice to have a paging tree view. One could probably adapt `csvlens` to do something like this.

**Comment by theAkito** [comment]
Is there a reason this is not operated on the natural structure type, like XML & JSON?
